File: Orbixtrade/websocketClient.go
package main

import (
	"fmt"
	"log"
	"os"
	"os/signal"
	"time"

	"github.com/gorilla/websocket"
)

func main() {
	interrupt := make(chan os.Signal, 1)
	signal.Notify(interrupt, os.Interrupt)

	url := "wss://www.orbixtrade.com/ws/stream?streams=!miniTicker@arr@3000ms"
	fmt.Printf("Connecting to %s\n", url)

	c, _, err := websocket.DefaultDialer.Dial(url, nil)
	if err != nil {
		log.Fatal("Dial error:", err)
	}
	defer c.Close()

	done := make(chan struct{})

	subscribeMessage := map[string]interface{}{
		"method": "SUBSCRIBE",
		"params": []string{
			//"btcthb@trade",
			"btc_thb@depth",
			"btc_thb@aggTrade",
		},
		"id": 2,
	}

	// Send the subscription message
	err = c.WriteJSON(subscribeMessage)
	if err != nil {
		log.Fatalf("Error sending subscription message: %v", err)
	}
	for {
		_, message, err := c.ReadMessage()
		if err != nil {
			log.Println("Read error:", err)
			return
		}
		// To do process message
		ProcessMsg(message)

	}

	ticker := time.NewTicker(time.Second)
	defer ticker.Stop()

	for {
		select {
		case <-done:
			return
		case t := <-ticker.C:
			err := c.WriteMessage(websocket.TextMessage, []byte(t.String()))
			if err != nil {
				log.Println("Write error:", err)
				return
			}
		case <-interrupt:
			log.Println("Interrupt received, closing connection")

			// Cleanly close the connection by sending a close message and then
			// waiting (with timeout) for the server to close the connection.
			err := c.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
			if err != nil {
				log.Println("Write close error:", err)
				return
			}
			select {
			case <-done:
			case <-time.After(time.Second):
			}
			return
		}
	}
}
func ProcessMsg(message []byte) {
	log.Printf("Received: %s", message)
}

File: Orbixtrade/websocket_client.py
import asyncio
import websockets
import signal
import json
import time

async def process_msg(message):
    print(f"Received: {message}")

async def main():
    url = "wss://www.orbixtrade.com/ws/stream?streams=!miniTicker@arr@3000ms"
    print(f"Connecting to {url}")

    async with websockets.connect(url) as websocket:
        subscribe_message = {
            "method": "SUBSCRIBE",
            "params": [
                # "btcthb@trade",
                "btc_thb@depth",
                "btc_thb@aggTrade",
            ],
            "id": 2,
        }

        # Send the subscription message
        await websocket.send(json.dumps(subscribe_message))

        async def read_messages():
            async for message in websocket:
                await process_msg(message)

        # Handle interrupts
        loop = asyncio.get_event_loop()
        stop = asyncio.Future()

        def handle_interrupt():
            if not stop.done():
                stop.set_result(None)

        loop.add_signal_handler(signal.SIGINT, handle_interrupt)

        # Send a message every second
        async def send_ticker():
            while not stop.done():
                await websocket.send(json.dumps({"type": "ping", "time": time.time()}))
                await asyncio.sleep(1)

        await asyncio.gather(read_messages(), send_ticker(), stop)

        # Cleanly close the connection
        close_message = json.dumps({"type": "close", "time": time.time()})
        await websocket.send(close_message)

if __name__ == "__main__":
    asyncio.run(main())
